File: app/src/androidTest/java/transfertests/ListTransferTest.java
package transfertests;

import com.klinker.android.twitter_l.data.sq_lite.ListSQLiteHelper;

import org.junit.After;
import org.junit.AfterClass;
import org.junit.Before;
import org.junit.BeforeClass;
import org.junit.Test;

import java.util.List;

public class ListTransferTest extends TransferTest {

    @BeforeClass
    public static void initDatabase() {
        String tableCreation = ListSQLiteHelper.DATABASE_CREATE;
        String addMediaLengthField = ListSQLiteHelper.DATABASE_ADD_MEDIA_LENGTH_FIELD;

        initSourceDatabase(tableCreation, addMediaLengthField);
        initTestDatabase();
    }


    @Test
    public void testBasicListTransfer() {

    }

    @Test
    public void testTransferIfNoSourceDatabase() {

    }


    @After
    public void clearDatabases() {
        clearTestDatabase();
        clearSourceDatabase(ListSQLiteHelper.TABLE_HOME);
    }

    @AfterClass
    public static void closeDatabase() {
        testDatabase.close();
        sourceDatabase.close();
    }
}

File: app/src/androidTest/java/transfertests/ListTransferTest.kt
package transfertests

import com.klinker.android.twitter_l.data.sq_lite.ListSQLiteHelper

import org.hamcrest.Description
import org.hamcrest.TypeSafeMatcher
import org.junit.After
import org.junit.AfterClass
import org.junit.Before
import org.junit.BeforeClass
import org.junit.Test

class ListTransferTest : TransferTest() {


    @Test
    fun testBasicListTransfer() {

    }

    @Test
    fun testTransferIfNoSourceDatabase() {

    }


    @After
    fun clearDatabases() {
        clearTestDatabase()
        clearSourceDatabase(ListSQLiteHelper.TABLE_HOME)
    }

    companion object {

        @BeforeClass
        fun initDatabase() {
            val tableCreation = ListSQLiteHelper.DATABASE_CREATE
            val addMediaLengthField = ListSQLiteHelper.DATABASE_ADD_MEDIA_LENGTH_FIELD

            TransferTest.initSourceDatabase(tableCreation, addMediaLengthField)
            TransferTest.initTestDatabase()
        }

        @AfterClass
        fun closeDatabase() {
            TransferTest.closeTestDatabase()
            TransferTest.closeSourceDatabase()
        }
    }
}


internal class MockList


internal class MockListMatcher private constructor(private val expected: MockList) : TypeSafeMatcher<MockList>() {

    override fun matchesSafely(item: MockList): Boolean {
        return false
    }

    override fun describeTo(description: Description) {

    }

    override fun describeMismatchSafely(item: MockList, description: Description) {

    }

    companion object {

        fun matchesTwitterList(expected: MockList): MockListMatcher {
            return MockListMatcher(expected)
        }
    }

}
